src/Plugin/EntityReferenceSelection/OgUserSelection.php

```diff
@@ -153,11 +153,13 @@ public function entityQueryAlter(SelectInterface $query) {
     }
 
     // Left join to the OG membership base table.
-    $query->leftJoin('og_membership', 'ogm', "base_table.uid = ogm.uid");
+    $query->leftJoin('og_membership', 'ogm', "base_table.uid = ogm.uid AND ogm.entity_type != :entity_type AND ogm.entity_id != :entity_id", [
```

Comment:
The conditions should be =, not !=. We want to attempt to join to the memberships for each user in the query, in the current group.